src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -1007,6 +1011,7 @@ public ResponseEntity<Void> addUserToCourseGroup(String userLogin, User instruct
                 throw new EntityNotFoundException("User", userLogin);
             }
             courseService.addUserToGroup(userToAddToGroup.get(), group, role);
+            channelService.registerUserToDefaultChannels(userToAddToGroup.get(), group, role);
```

Comment:
this could take some time and it is typically not needed directly. Therefore, we should execute this asynchronously to avoid waiting time for the end user.
In particular when we have many exercise and lecture channels, this operation might be quite slow

Reply:
made it `async`